lightningd/notification.c

```diff
@@ -59,4 +60,17 @@ void notify_warning(struct lightningd *ld, struct log_entry *l)
 	json_object_end(n->stream); /* .warning */
 	jsonrpc_notification_end(n);
 	plugins_notify(ld->plugins, take(n));
-}
\ No newline at end of file
+}
+
+void notify_invoice_payment(struct lightningd *ld, struct amount_msat amount,
+			    struct preimage preimage, const struct json_escape *label)
+{
+	struct jsonrpc_notification *n =
+		jsonrpc_notification_start(NULL, notification_topics[3]);
+	json_add_string(n->stream, "msat",
```

Comment:
Might be better to wrap this in an object, otherwise the pylightning code will need to update in sync with the changes to the arguments (it unpacks the arguments into a function call).